include/libopencm3/stm32/g4/rcc.h

```diff
@@ -691,242 +751,303 @@ extern uint32_t rcc_apb2_frequency;
 
 /* --- Function prototypes ------------------------------------------------- */
 
-enum rcc_osc {
-	RCC_PLL, RCC_HSE, RCC_HSI16, RCC_LSE, RCC_LSI, RCC_HSI48
+enum rcc_clock_3v3 {
+	RCC_CLOCK_3V3_24MHZ,
+	RCC_CLOCK_3V3_48MHZ,
+	RCC_CLOCK_3V3_96MHZ,
+	RCC_CLOCK_3V3_170MHZ,
+	RCC_CLOCK_3V3_END
 };
 
+struct rcc_clock_scale {
+	uint8_t pllm;
+	uint16_t plln;
+	uint8_t pllp;
+	uint8_t pllq;
+	uint8_t pllr;
+	uint8_t pll_source;
+	uint8_t hpre;
+	uint8_t ppre1;
+	uint8_t ppre2;
+	enum pwr_vos_scale vos_scale;
+	bool boost;
+	uint32_t flash_config;
+	uint8_t flash_waitstates;
+	uint32_t ahb_frequency;
+	uint32_t apb1_frequency;
+	uint32_t apb2_frequency;
+};
+
+extern const struct rcc_clock_scale rcc_hsi_configs[RCC_CLOCK_3V3_END];
+extern const struct rcc_clock_scale rcc_hse_8mhz_3v3[RCC_CLOCK_3V3_END];
+extern const struct rcc_clock_scale rcc_hse_12mhz_3v3[RCC_CLOCK_3V3_END];
+extern const struct rcc_clock_scale rcc_hse_16mhz_3v3[RCC_CLOCK_3V3_END];
```

Comment:
This is it, the clock scale struct, that's all that's new in this file, apart from the ~standard function declarations. I don't buy it at all that this gets 4 new authros and a 2009 date when it's just getting definititions that are "compatible" with another file, and I _certainly_ don't think that the massive whitespace changes are useful in this commit, if ever.

Reply:
This should be fixed now.